pennylane/labs/resource_estimation/ops/op_math/controlled_ops.py

```diff
@@ -20,24 +20,435 @@
+                "num_ctrl_values": num_ctrl_values,
+                "num_work_wires": num_work_wires,
+            },
+        )
+
+
+class ResourceCRX(qml.CRX, re.ResourceOperator):
+    r"""Resource class for CRX
+
+    Resources:
+
+        The resources are derived from the following identities:
 
         .. math::
 
-            CR_\phi(\phi) = \begin{bmatrix}
-                    1 & 0 & 0 & 0 \\
-                    0 & 1 & 0 & 0 \\
-                    0 & 0 & 1 & 0 \\
-                    0 & 0 & 0 & e^{i\phi}
-                \end{bmatrix} =
-            (R_\phi(\phi/2) \otimes I) \cdot CNOT \cdot (I \otimes R_\phi(-\phi/2)) \cdot CNOT \cdot (I \otimes R_\phi(\phi/2))
+            \begin{align}
+                \hat{RZ}(- \theta) = \hat{X} \dot \hat{RZ}(\theta) \dot \hat{X}, \\
+                \hat{X} &= \hat{H} \dot \hat{Z}  \dot \hat{H}
+            \end{align}
+
+        The expression for controlled-RZ gates is used as defined in (figure 1b.) the paper
+        `T-count and T-depth of any multi-qubit unitary <https://arxiv.org/pdf/2110.10292>`_.
+    """
+
+    @staticmethod
+    def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        gate_types = {}
+
+        h = re.ResourceHadamard.resource_rep()
+        rz = re.ResourceRZ.resource_rep()
+        cnot = re.ResourceCNOT.resource_rep()
+
+        gate_types[cnot] = 2
+        gate_types[rz] = 2
+        gate_types[h] = 2
+
+        return gate_types
+
+    def resource_params(self) -> dict:
+        return {}
+
+    @classmethod
+    def resource_rep(cls) -> re.CompressedResourceOp:
+        return re.CompressedResourceOp(cls, {})
+
+
+class ResourceCRY(qml.CRY, re.ResourceOperator):
+    r"""Resource class for CRY
 
+    Resources:
+        The resources are derived from (in figure 1b.) the paper `T-count and T-depth of any multi-qubit
+        unitary <https://arxiv.org/pdf/2110.10292>`_. The resources are derived with the following identity:
+
+        .. math:: \hat{RY}(- \theta) = \hat{X} \dot \hat{RY}(\theta) \dot \hat{X}.
+
+    """
+
+    @staticmethod
+    def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        gate_types = {}
+
+        cnot = re.ResourceCNOT.resource_rep()
+        ry = re.ResourceRY.resource_rep()
+
+        gate_types[cnot] = 2
+        gate_types[ry] = 2
+
+        return gate_types
+
+    def resource_params(self) -> dict:
+        return {}
+
+    @classmethod
+    def resource_rep(cls) -> re.CompressedResourceOp:
+        return re.CompressedResourceOp(cls, {})
+
+
+class ResourceCRZ(qml.CRZ, re.ResourceOperator):
+    r"""Resource class for CRZ
+
+    Resources:
+        The resources are obtained from (in figure 1b.) the paper `T-count and T-depth of any multi-qubit
+        unitary <https://arxiv.org/pdf/2110.10292>`_. The resources are derived from the following identity:
+
+        .. math:: \hat{RZ}(- \theta) = \hat{X} \dot \hat{RZ}(\theta) \dot \hat{X}.
```

Comment:
![Screenshot from 2024-12-02 09-57-28](https://github.com/user-attachments/assets/a7e7ddc9-7f4d-4cc9-aa4e-c7fb7980fecc)

Check that this renders properly.

Reply:
Should be fixed now 👍🏼